src/PureCloud.Client/Apis/SettingsApi.cs

```diff
@@ -12,152 +10,164 @@ namespace PureCloud.Client.Apis;
 
-        var response = await _httpClient.PatchAsJsonAsync(uri, body, _options, cancellationToken);
+        var response = await client.PatchAsJsonAsync("api/v2/emails/settings", body, _options.JsonSerializerOptions, cancellationToken);
 
         response.EnsureSuccessStatusCode();
 
-        return await response.Content.ReadFromJsonAsync<EmailSettings>(_options, cancellationToken);
+        return await response.Content.ReadFromJsonAsync<EmailSettings>(_options.JsonSerializerOptions, cancellationToken);
     }
 
     /// <inheritdoc />
-
     public async Task<EmailThreadingSettings> GetEmailsSettingsThreadingAsync(CancellationToken cancellationToken = default)
     {
-        var uri = UriHelper.GetUri("api/v2/emails/settings/threading", null);
+        var client = _httpClientFactory.CreateClient(PureCloudConstants.PureCloudClientName);
 
-        var response = await _httpClient.GetAsync(uri, cancellationToken);
+        var response = await client.GetAsync("api/v2/emails/settings/threading", cancellationToken);
 
         response.EnsureSuccessStatusCode();
 
-        return await response.Content.ReadFromJsonAsync<EmailThreadingSettings>(_options, cancellationToken);
+        return await response.Content.ReadFromJsonAsync<EmailThreadingSettings>(_options.JsonSerializerOptions, cancellationToken);
     }
 
     /// <inheritdoc />
-
     public async Task<EmailThreadingSettings> UpdateEmailsSettingsThreadingAsync(EmailThreadingSettings body, CancellationToken cancellationToken = default)
     {
         ArgumentNullException.ThrowIfNull(body);
 
-        var uri = UriHelper.GetUri("api/v2/emails/settings/threading", null);
+        var client = _httpClientFactory.CreateClient(PureCloudConstants.PureCloudClientName);
 
-        var response = await _httpClient.PatchAsJsonAsync(uri, body, _options, cancellationToken);
+        var response = await client.PatchAsJsonAsync("api/v2/emails/settings/threading", body, _options.JsonSerializerOptions, cancellationToken);
 
         response.EnsureSuccessStatusCode();
 
-        return await response.Content.ReadFromJsonAsync<EmailThreadingSettings>(_options, cancellationToken);
+        return await response.Content.ReadFromJsonAsync<EmailThreadingSettings>(_options.JsonSerializerOptions, cancellationToken);
     }
 
     /// <inheritdoc />
-
-    public async Task DeleteEmailsSettingsThreadingAsync(CancellationToken cancellationToken = default)
+    public async Task<bool> DeleteEmailsSettingsThreadingAsync(CancellationToken cancellationToken = default)
     {
-        var uri = UriHelper.GetUri("api/v2/emails/settings/threading", null);
+        var client = _httpClientFactory.CreateClient(PureCloudConstants.PureCloudClientName);
 
-        var response = await _httpClient.DeleteAsync(uri, cancellationToken);
+        var response = await client.DeleteAsync("api/v2/emails/settings/threading", cancellationToken);
 
         response.EnsureSuccessStatusCode();
```

Comment:
When returning Boolean, EnsureSuccessStateCode should not be invoked. Apply this everywhere